src/api/app/controllers/attribute_controller.rb

```diff
@@ -2,14 +2,16 @@ class AttributeController < ApplicationController
   include ValidationHelper
 
   validate_action index: { method: :get, response: :directory }
-  validate_action namespace_definition: { method: :get, response: :attribute_namespace_meta }
-  validate_action namespace_definition: { method: :delete, response: :status }
-  validate_action namespace_definition: { method: :put, request: :attribute_namespace_meta, response: :status }
-  validate_action namespace_definition: { method: :post, request: :attribute_namespace_meta, response: :status }
-  validate_action attribute_definition: { method: :get, response: :attrib_type }
-  validate_action attribute_definition: { method: :delete, response: :status }
-  validate_action attribute_definition: { method: :put, request: :attrib_type, response: :status }
-  validate_action attribute_definition: { method: :post, request: :attrib_type, response: :status }
+  validate_action show_namespace_definition: { method: :get, response: :attribute_namespace_meta }
+  validate_action delete_namespace_definition: { method: :delete, response: :status }
+  validate_action update_namespace_definition: { method: :put, request: :attribute_namespace_meta, response: :status }
+  validate_action update_namespace_definition: { method: :post, request: :attribute_namespace_meta, response: :status }
+  validate_action show_attribute_definition: { method: :get, response: :attrib_type }
+  validate_action delete_attribute_definition: { method: :delete, response: :status }
+  validate_action update_attribute_definition: { method: :put, request: :attrib_type, response: :status }
+  validate_action update_attribute_definition: { method: :post, request: :attrib_type, response: :status }
```

Comment:
All of this sounds very much to me like two controllers, a `attribute_namespace` and a `attribute` controller.

And while you're at it, can you get rid of this stupid `_definition` prefix? :-)

Reply:
It's actually 3 controllers - you notice when you try to remove the _definition suffix (prefix is ... anyway)